app/views/session_timeout/_warning.html.erb

```diff
@@ -1,24 +1,22 @@
-<div id='session-timeout-msg' class='display-none' tabindex='-1'>
-  <%= render layout: 'shared/modal_layout', locals: { role: 'dialog' } do |label_id, description_id| %>
-    <div class='padding-4 tablet:padding-6 cntnr-xxskinny bg-white rounded-xxl modal-timeout'>
-      <h2 class='margin-y-2 fs-20p text-normal text-center' id='<%= label_id %>'>
-        <%= t('headings.session_timeout_warning') %>
-      </h2>
-      <hr class='margin-bottom-4 border-width-05' />
-      <div class='margin-bottom-6'>
-        <p>
-          <%= modal.message %>
-        </p>
-        <p class="usa-sr-only" id='<%= description_id %>' role="timer" aria-live="polite" aria-atomic="true">
-          <%= modal.sr_message %>
-        </p>
-      </div>
-      <%= button_tag modal.continue,
-                     id: 'session-keepalive-btn',
-                     class: 'usa-button usa-button--big usa-button--full-width margin-bottom-2' %>
-      <%= link_to modal.sign_out,
-                  destroy_user_session_path,
-                  class: 'usa-button usa-button--big usa-button--full-width usa-button--outline' %>
+<%= render ModalComponent.new(class: 'session-timeout') do |c| %>
+  <div class='padding-4 tablet:padding-6 cntnr-xxskinny bg-white rounded-xxl modal-timeout'>
+    <h2 class='margin-y-2 fs-20p text-normal text-center' id='<%= c.label_id %>'>
+      <%= t('headings.session_timeout_warning') %>
+    </h2>
+    <hr class='margin-bottom-4 border-width-05' />
+    <div class='margin-bottom-6'>
+      <p>
+        <%= modal.message %>
```

Comment:
so we pass in `modal` as a local but then we have the local `c` as the other modal, what if we renamed these?

`modal` becomes `modal_content`

and `c` becomes `modal_component`?

Reply:
I'm also not typically in favor of ambiguous abbreviations, but the `c` thing was something I picked up from the [ViewComponent slot documentation](https://viewcomponent.org/guide/slots.html) and use already in a few places. Generally open to expanding if it'd help clarify, but may also want to do so consistently in other components.

Reply:
maybe if we rename `modal` that would resolve the confusion, I was just surprised to see us pass in `modal` but then instantiate a `ModalComponent` separately

Reply:
Yeah, that works. It didn't occur to me since I was just preserving what had existed already. Looking a bit closer, I think maybe `modal_presenter` might fit as a good name, since it'll be either an instance of [`FullySignedInModalPresenter`](https://github.com/18F/identity-idp/blob/main/app/presenters/fully_signed_in_modal_presenter.rb) or [`PartiallySignedInModalPresenter`](https://github.com/18F/identity-idp/blob/main/app/presenters/partially_signed_in_modal_presenter.rb).

Reply:
Renamed in a0f9db610.